src/services/analytics/tx-tracking.ts

```diff
@@ -0,0 +1,68 @@
+import { TX_TYPES } from '@/services/analytics/events/transactions'
+import { getTxDetails } from '@/services/tx/txDetails'
+import { isWalletConnectSafeApp } from '@/services/walletconnect/utils'
+import { SettingsInfoType, type TransactionDetails } from '@safe-global/safe-gateway-typescript-sdk'
+import {
+  isERC721Transfer,
+  isMultiSendTxInfo,
+  isSettingsChangeTxInfo,
+  isTransferTxInfo,
+  isCustomTxInfo,
+  isCancellationTxInfo,
+} from '@/utils/transaction-guards'
+
+export const getTransactionTrackingType = async (txId: string, chainId: string): Promise<string | undefined> => {
+  let details: TransactionDetails
+  try {
+    details = await getTxDetails(txId, chainId)
+  } catch {
+    return TX_TYPES.custom
+  }
+  const { txInfo } = details
+
+  if (isTransferTxInfo(txInfo)) {
+    if (isERC721Transfer(txInfo.transferInfo)) {
+      return TX_TYPES.transfer_nft
+    }
+    return TX_TYPES.transfer_token
+  }
+
+  if (isSettingsChangeTxInfo(txInfo)) {
+    switch (txInfo.settingsInfo?.type) {
+      case SettingsInfoType.ADD_OWNER: {
+        return TX_TYPES.owner_add
+      }
+      case SettingsInfoType.REMOVE_OWNER: {
+        return TX_TYPES.owner_remove
+      }
+      case SettingsInfoType.SWAP_OWNER: {
+        return TX_TYPES.owner_swap
+      }
+      case SettingsInfoType.CHANGE_THRESHOLD: {
+        return TX_TYPES.owner_threshold_change
+      }
+      case SettingsInfoType.DISABLE_MODULE: {
+        return TX_TYPES.module_remove
+      }
+      case SettingsInfoType.DELETE_GUARD: {
+        return TX_TYPES.guard_remove
+      }
+    }
+  }
+
+  if (isCustomTxInfo(txInfo)) {
+    if (isCancellationTxInfo(txInfo)) {
+      return TX_TYPES.rejection
+    }
+
+    if (details.safeAppInfo) {
+      return isWalletConnectSafeApp(details.safeAppInfo.url) ? TX_TYPES.walletconnect : TX_TYPES.safeapps
```

Comment:
I feel like we are mixing source of transaction (SafeApp, WalletConnect) and transaction type here.
Should we add the source as label?

Reply:
I’ll double check.

Reply:
Actually, it's only custom if it's a multisend or some other custom contract interaction. For single ERC20 transfers, it's categorized as a transfer. So you're right it's mixing apples and oranges a bit, but it will happen only within the custom category. I think it's an OK solution w/o introducing a new custom dimension.